site/src/components/Search/SearchField.tsx

```diff
@@ -0,0 +1,83 @@
+import { useTheme } from "@emotion/react";
+import CloseOutlined from "@mui/icons-material/CloseOutlined";
+import SearchOutlined from "@mui/icons-material/SearchOutlined";
+import IconButton from "@mui/material/IconButton";
+import InputAdornment from "@mui/material/InputAdornment";
+import TextField, { type TextFieldProps } from "@mui/material/TextField";
+import Tooltip from "@mui/material/Tooltip";
+import { visuallyHidden } from "@mui/utils";
+import type { FC } from "react";
+
+export type SearchFieldProps = Omit<
+  TextFieldProps,
+  "onChange" | "id" | "label" | "value" | "error"
+> & {
+  id: string;
+  label: string;
+  value: string;
+  autoFocus?: boolean;
+  error?: string;
+  onChange: (value: string) => void;
+};
+
+export const SearchField: FC<SearchFieldProps> = (props) => {
+  const theme = useTheme();
+  const { value, label, id, error, onChange, autoFocus, ...textFieldProps } =
+    props;
+  const isEmpty = value.length === 0;
+
+  return (
+    <>
+      <label htmlFor={id} css={{ ...visuallyHidden }}>
+        {label}
+      </label>
+      <TextField
+        {...textFieldProps}
+        error={Boolean(error)}
+        helperText={error}
+        type="text"
+        InputProps={{
+          ...textFieldProps.InputProps,
+          autoFocus,
+          id,
+          size: "small",
+          startAdornment: (
+            <InputAdornment position="start">
+              <SearchOutlined
+                role="presentation"
+                css={{
+                  fontSize: 14,
+                  color: theme.palette.text.secondary,
+                }}
+              />
+            </InputAdornment>
+          ),
+          endAdornment: !isEmpty && (
+            <Tooltip title="Clear filter">
+              <IconButton
+                size="small"
+                onClick={() => {
+                  onChange("");
+                }}
+              >
+                <CloseOutlined
+                  css={{
+                    fontSize: 14,
+                    color: theme.palette.text.secondary,
+                  }}
+                />
+                <span css={{ ...visuallyHidden }}>Clear filter</span>
+              </IconButton>
+            </Tooltip>
+          ),
+        }}
+        value={value}
+        onChange={(e) => {
+          onChange(e.currentTarget.value);
```

Comment:
Honestly, I didn't know the difference. I was using `currentTarget` because I see people using it in tutorials and docs. I read about it here https://stackoverflow.com/questions/10086427/what-is-the-exact-difference-between-currenttarget-property-and-target-property and I'm good by using target 👍 Thanks for bringing this up